R/input-check-helpers.R

```diff
@@ -0,0 +1,207 @@
+    if (isFALSE(all(true_values %in% c(0, 1)))) {
+      stop("For a binary forecast, all true_values should be either 0 or 1.")
+    }
+  }
+}
+
+
+#' @title Check Variable is not NULL
+#'
+#' @description
+#' Check whether a certain variable is not `NULL` and return the name of that
+#' variable and the function call where the variable is missing. This function
+#' is a helper function that should only be called within other functions
+#' @param ... The variables to check
+#' @return The function returns `NULL`, but throws an error if the variable is
+#' missing.
+#'
+#' @keywords internal
+check_not_null <- function(...) {
+  vars <- list(...)
+  varnames <- names(vars)
+
+  for (i in seq_along(vars)) {
+    varname <- varnames[i]
+    if (is.null(vars[[i]])) {
+      calling_function <- deparse1(sys.calls()[[sys.nframe() - 1]])
+      stop(paste0(
+        "variable '", varname,
+        "' is `NULL` in the following function call: '",
+        calling_function, "'"
+      ))
+    }
+  }
+  return(invisible(NULL))
+}
+
+
+#' @title Check Length
+#'
+#' @description
+#' Check whether variables all have the same length
+#' @param ... The variables to check
+#' @param one_allowed logical, allow arguments of length one that can be
+#' recycled
+#'
+#' @return The function returns `NULL`, but throws an error if variable lengths
+#' differ
+#'
+#' @keywords internal
+check_equal_length <- function(...,
+                               one_allowed = TRUE) {
+  vars <- list(...)
+  lengths <- sapply(vars,
+    FUN = function(x) {
+      length(x)
+    }
+  )
+
+  lengths <- unique(lengths)
+
+  if (one_allowed) {
+    lengths <- lengths[lengths != 1]
+  }
+
+  if (length(unique(lengths)) != 1) {
+    calling_function <- deparse1(sys.calls()[[sys.nframe() - 1]])
+    stop(paste0(
+      "Arguments passed to the following function call: '",
+      calling_function,
+      "' should have the same length (or length one). Arguments have the following lengths: ",
+      paste0(lengths, collapse = ", ")
+    ))
+  }
+  return(invisible(NULL))
+}
+
+
+
+#' @title Check whether the desired metrics are available in scoringutils
+#'
+#' @description Helper function to check whether desired metrics are
+#' available. If the input is `NULL`, all metrics will be returned.
+#'
+#' @param metrics character vector with desired metrics
+#'
+#' @return A character vector with metrics that can be used for downstream
+#' computation
+#'
+#' @keywords internal
+
+check_metrics <- function(metrics) {
+  # use all available metrics if none are given
+  if (is.null(metrics)) {
+    metrics <- available_metrics()
+  }
+
+  # check desired metrics are actually available in scoringutils
+  available_metrics <- available_metrics()
+  if (!all(metrics %in% available_metrics)) {
+    msg <- paste(
+      "The following metrics are not available:",
+      paste(setdiff(metrics, available_metrics), collapse = ", ")
+    )
+    warning(msg)
+  }
+  return(metrics)
+}
```

Comment:
This could probably be replaced by `match.arg(metrics, available_metrics, several.ok = TRUE)`